### DISCUSSION TITLE
Sorting is strange

### DISCUSSION THREAD

**Discussion by guettli** [djangopackages/djangopackages#664] [Grids]
**Describe the bug**

The sorting is strange:

![image](https://user-images.githubusercontent.com/414336/115868831-528f1080-a43d-11eb-82e1-1ffbdde5fae5.png)

Why is the outdated package almost on top?

**To Reproduce**

1. Go to https://djangopackages.org/grids/g/payment-processing/
2. Look into the second column.
3. Look at the last release date: Several years old.

**Expected behavior**

I want to see maintained packages on top.

**Comment by juanbeniteza** [comment]
I guess it has something to do with the number of stars and forks that package has.

**Comment by jefftriplett** [comment]
This will be partially addressed in #595, but overall is going to need several small updates to fix it. As of today, the primary metric used to sort packages on a grid is `repo_watchers` which is why an outdated package never drops off.

Obviously this is an issue. In #681 we want to fix deprecated packages and add support in our Package model for marking them so we can sort them out. I also think we want to make it easier (fewer joins) to track how long it's been since a package was either released and/or updated via commits. Some of this code might favor GitHub though, so I want to make sure that being on GitHub isn't an arbitrary boost on whatever scoring system we adopt. 

Another thing that sticks out to me is that we aren't using the features as a scoring/ranking factor for our grids which I also want to address. This is going to take the longest to address, but once we can depreciate packages and reasonably identify what is out-of-date, it's not a bad project to take on next.

**Comment by jefftriplett** [comment]
@guettli this is partially resolved now: https://djangopackages.org/grids/g/payment-processing/

We have made several baby steps and have more to do (probably after DjangoCon US). I'd like for the popularity of the package to be a factor, but obviously not the only factor. 

I'd also like to look at adopting htmlx where it makes sense. We have just enough javascript that hasn't been touched in a while, that I suspect we have avoided features as a result. I'm not the best at JS, but I also want an excuse to use htmlx 😄

**Reply by guettli** [reply]
Thank you! The new sorting looks much better.
